quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -234,14 +319,75 @@ public Optional<DeprecatedMetadata> getDeprecatedMetadata() {
         return option.getDeprecatedMetadata();
     }
 
+    /**
+     * An option is considered a wildcard option if its key contains a wildcard placeholder (e.g. log-level-<category>).
+     * The placeholder must be denoted by the '<' and '>' characters.
+     */
+    public boolean hasWildcard() {
+        return fromWildcardPattern != null;
+    }
+
+    /**
+     * Checks if the given option name matches the wildcard pattern of this option.
+     * E.g. check if "log-level-io.quarkus" matches the wildcard pattern "log-level-<category>".
+     */
+    public boolean matchesWildcardOptionName(String name) {
+        if (!hasWildcard()) {
+            return false;
+        }
+        return fromWildcardPattern.matcher(name).matches() || envVarNameWildcardPattern.matcher(name).matches()
+                || (toWildcardPattern != null && toWildcardPattern.matcher(name).matches());
+    }
+
+    /**
+     * Extracts the wildcard value from the given option name.
+     * E.g. for the option "log-level-<category>" and the option name "log-level-io.quarkus",
+     * the wildcard value would be "io.quarkus".
+     */
+    private Optional<String> getWildcardValue(String option, boolean includeMappedToOptions) {
+        if (!hasWildcard()) {
+            return Optional.empty();
+        }
+
+        Matcher matcher = fromWildcardPattern.matcher(option);
+        if (matcher.matches()) {
+            return Optional.of(matcher.group(1));
+        }
+
+        matcher = envVarNameWildcardPattern.matcher(option);
```

Comment:
Does this need a check that the option is actually coming from the env? Otherwise it seems like we'll start allowing env style keys in places like the keycloak.conf.

Reply:
Good point, added additional toggle for checking env vars.